lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
+  attr_reader :name, :address, :phone,
+              :services, :registered_vehicles, :collected_fees
+              :administer_written_test
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @administer_written_test
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless services.include?("Vehicle Registration")
+    vehicle.register
+    vehicle.registration_date == Date.today
+    @collected_fees += fee_amount(vehicle)
+     @registered_vehicles << vehicle
+  end
+  
+  def fee_amount(vehicle)
+    if vehicle.antique?
+      25
+    elsif vehicle.electric_vehicle?
+      200
+    else
+      100
+    end
+  end
+  
+  def administer_written_test(registrant)
+    return false unless services.include?("Written Test")
```

Comment:
I would add a space after this guard clause so it's really clear that it's one line of code and if it's not activated, then the rest of the block will run.